PhotoAlbumSample/app/src/androidTest/java/com/amazonaws/android/samples/photosharing/AppSyncMutationUITest.java

```diff
@@ -0,0 +1,222 @@
+import static org.hamcrest.Matchers.allOf;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class AppSyncMutationUITest {
+
+    private static final int MAX_TIME_OUT = 60 * 1000;
+    private final static String ALBUM_NAME_FOR_TESTING = "TestAppSync";
+    private final static String TAG = AppSyncMutationUITest.class.getSimpleName();
+
+    @Rule
+    public ActivityTestRule<LoginActivity> mActivityTestRule = new ActivityTestRule<>(LoginActivity.class);
+
+    @Before
+    public void setUp() throws Exception {
+        Context appContext = InstrumentationRegistry.getTargetContext();
+
+        final CountDownLatch latch = new CountDownLatch(1);
+        AWSMobileClient.getInstance().initialize(appContext, new Callback<UserStateDetails>() {
+            @Override
+            public void onResult(UserStateDetails result) {
+                latch.countDown();
+            }
+
+            @Override
+            public void onError(Exception e) {
+                latch.countDown();
+            }
+        });
+        latch.await();
+
+        final AWSConfiguration awsConfiguration = AWSMobileClient.getInstance().getConfiguration();
+
+        // AWS Configuration file should have AppSync configured.
+        // Should be generated by Amplify CLI.
+        assertNotNull(awsConfiguration.optJsonObject("AppSync"));
+        Log.e(TAG, "AppSync has been configured.");
+
+        // Check if goes back to LoginActivity
+        int timeOut = 0;
+        while (timeOut < MAX_TIME_OUT) {
+            try {
+                ViewInteraction button2 = onView(allOf(withId(R.id.user_pool_sign_in_view_id)));
+                button2.check(matches(isDisplayed()));
+                Log.e(TAG,"The view has gone back to LoginActivity.");
+                break;
+            } catch (NoMatchingViewException e) {
+            }
+
+            Thread.sleep(5000);
+            timeOut += 5000;
+        }
+
+        // Do sign in first
+        UIActionsUtil.signIn(UIActionsUtil.getUsername(), UIActionsUtil.getPassword());
+
+        // Check if user successfully signed in.
+        AWSMobileClient.getInstance().confirmSignIn("signInChallengeResponse", new Callback<SignInResult>() {
+            @Override
+            public void onResult(SignInResult result) {
+                Log.e(TAG, "SignInResult: " + result);
+                assertEquals(result.getSignInState(), SignInState.DONE);
+            }
+
+            @Override
+            public void onError(Exception e) {
+                Log.e(TAG, "SignInResult Error: " + e);
+            }
+        });
+
+        // Check if it jumps into AlbumActivity
+        timeOut = 0;
+        while (timeOut < MAX_TIME_OUT) {
+            try {
+                onView(allOf(withText("AlbumActivity"), childAtPosition(allOf(withId(R.id.action_bar), childAtPosition(
```

Comment:
Do not hardcode the activity name here. Move it to a constant or reference it directly by saying `AlbumActivity.class.getSimpleName();`